src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/SerilogLoggerProvider.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Serilog;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Internal
+{
+    /// <summary>
+    /// Represents a Serilog logger provider use for Azure WebApp.
+    /// </summary>
+    public abstract class SerilogLoggerProvider : ILoggerProvider
+    {
+        // Solution suggested by the Serilog creator http://stackoverflow.com/questions/30849166/how-to-turn-off-serilog
+        /// <summary>
+        /// The log level at which the logger is disabled.
+        /// </summary>
+        protected static LogEventLevel LogLevelDisabled = ((LogEventLevel)1 + (int)LogEventLevel.Fatal);
+
+        private readonly LoggingLevelSwitch _levelSwitch = new LoggingLevelSwitch();
+
+        private readonly IWebAppLogConfigurationReader _configReader;
+        private readonly ILoggerFactory _loggerFactory;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="SerilogLoggerProvider"/> class.
+        /// </summary>
+        /// <param name="configReader">The configuration reader</param>
+        /// <param name="configureLogger">The actions required to configure the logger</param>
+        public SerilogLoggerProvider(IWebAppLogConfigurationReader configReader, Action<LoggerConfiguration, WebAppLogConfiguration> configureLogger)
+        {
+            if (configReader == null)
+            {
+                throw new ArgumentNullException(nameof(configReader));
+            }
+            if (configureLogger == null)
+            {
+                throw new ArgumentNullException(nameof(configureLogger));
+            }
+
+            _configReader = configReader;
+            var webAppsConfiguration = configReader.Latest;
+
+            configReader.OnConfigurationChanged += OnConfigurationChanged;
+
+            var loggerConfiguration = new LoggerConfiguration()
+                .MinimumLevel.ControlledBy(_levelSwitch);
+            configureLogger(loggerConfiguration, webAppsConfiguration);
+            var serilogLogger = loggerConfiguration.CreateLogger();
+            
+            OnConfigurationChanged(webAppsConfiguration);
+
+            _loggerFactory = new LoggerFactory();
```

Comment:
Easy change for us, unless there's a catch I haven't spotted we should be able to make the `SerilogLogger` public in a point release in a few days.